src/internal/_objectAssign.js

```diff
@@ -0,0 +1,24 @@
+var _has = require('./_has');
+
+// Based on https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Object/assign
+module.exports = function _objectAssign(target) {
+  if (target === undefined || target === null) {
+    throw new TypeError('Cannot convert undefined or null to object');
+  }
```

Comment:
Thought you'd say that :) I left it in because that's the exception native Object.assign throws. I can leave it out of course.